### PR TITLE
Properly checks if api calls error out  (AIC-594)

### PR DESCRIPTION
**Pull request by sam33rdhakal**
* App now allows a user to proceed to the home screen irrespective of any exception type (when cached data is available) that happens during data fetching process.
* Updates the `getEvents` & `getExhibitions` method to properly check for error response.

### CODE REVIEW

File: app/src/main/kotlin/edu/artic/CrashlyticsTree.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
+
+        if (reportableLogs.contains(priority)) {
+            return
+        }
+
+        Crashlytics.setInt(CRASHLYTICS_KEY_PRIORITY, priority)
+        Crashlytics.setString(CRASHLYTICS_KEY_TAG, tag)
+        Crashlytics.setString(CRASHLYTICS_KEY_MESSAGE, message)
```
Comment (Cliabhach) [inline app/src/main/kotlin/edu/artic/CrashlyticsTree.kt]: Please just use the three-args `::log` method.

File: db/src/main/kotlin/edu/artic/db/ApiModule.kt [hunk 2]
```diff
@@ -51,7 +51,8 @@ abstract class ApiModule {
                 exhibitionDao: ArticExhibitionDao,
                 objectDao: ArticObjectDao,
                 articMapFloorDao: ArticMapFloorDao,
-                searchSuggestionDao: ArticSearchObjectDao
+                searchSuggestionDao: ArticSearchObjectDao,
+                appDaPrefManager: AppDataPreferencesManager
```
Comment (Cliabhach) [inline db/src/main/kotlin/edu/artic/db/ApiModule.kt]: `appDaPrefManager` -> `appDataPrefManager`

File: db/src/main/kotlin/edu/artic/db/RetrofitAppDataServiceProvider.kt [hunk 3]
```diff
@@ -87,37 +91,41 @@ class RetrofitAppDataServiceProvider(
...
+                        var url = dataObject.dataApiUrl + dataObject.exhibitionsEndpoint
+                        if (!url.contains("/search")) {
+                            url += "/search"
+                        }
+                        url += "?limit=99"
+
+                        val postParams = ApiBodyGenerator.createExhibitionQueryBody()
+
+                        service.getExhibitions(EXHIBITIONS_HEADER_ID, url, postParams)
+                                .map { it.throwIfError() }
```
Comment (Cliabhach) [inline db/src/main/kotlin/edu/artic/db/RetrofitAppDataServiceProvider.kt]: Please use method reference instead.

File: db/src/main/kotlin/edu/artic/db/RetrofitAppDataServiceProvider.kt [hunk 4]
```diff
@@ -128,36 +136,39 @@ class RetrofitAppDataServiceProvider(
...
+                        var url = dataObject.dataApiUrl + dataObject.eventsEndpoint
+                        if (!url.contains("/search")) {
+                            url += "/search"
                         }
+                        url += "?limit=500"
+
+                        val postParams = ApiBodyGenerator.createEventQueryBody()
+
+                        service.getEvents(EVENT_HEADER_ID, url, postParams)
+                                .map { it.throwIfError() }
```
Comment (Cliabhach) [inline db/src/main/kotlin/edu/artic/db/RetrofitAppDataServiceProvider.kt]: This one too.

File: app/src/main/kotlin/edu/artic/CrashlyticsTree.kt (hunk 1 above)
Comment (sam33rdhakal) [reply app/src/main/kotlin/edu/artic/CrashlyticsTree.kt]: We dont need it. Removed.

File: db/src/main/kotlin/edu/artic/db/ApiModule.kt (hunk 2 above)
Comment (sam33rdhakal) [reply db/src/main/kotlin/edu/artic/db/ApiModule.kt]: Updated.

File: db/src/main/kotlin/edu/artic/db/RetrofitAppDataServiceProvider.kt (hunk 3 above)
Comment (sam33rdhakal) [reply db/src/main/kotlin/edu/artic/db/RetrofitAppDataServiceProvider.kt]: Good idea. Updated.

File: db/src/main/kotlin/edu/artic/db/RetrofitAppDataServiceProvider.kt (hunk 4 above)
Comment (sam33rdhakal) [reply db/src/main/kotlin/edu/artic/db/RetrofitAppDataServiceProvider.kt]: Handled.

File: splash/src/main/kotlin/edu/artic/splash/SplashViewModel.kt [hunk 5]
```diff
@@ -66,8 +65,10 @@ class SplashViewModel @Inject constructor(
                 .disposedBy(disposeBag)
     }
 
-    
-    fun onNetworkErrorDialogDismissed() {
+    /**
+     * Allow user to proceed forward if app has cache data.
+     */
+    fun proceedToWelcomePageIfDataAvailable() {
```
Comment (Cliabhach) [inline splash/src/main/kotlin/edu/artic/splash/SplashViewModel.kt]: I like this name change.

**Pull request merged**